src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react';
+
+export const useDebounce = <Val>(value: Val, delay: number): Val => {
```

Comment:
Maybe default value? We will probably use the same value 95% of the times.

Reply:
Agree, will add it